sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -1205,6 +1205,32 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSparkSession
       }
     }
   }
+
+  test("SPARK-34167: read LongDecimals with precision < 10, VectorizedReader off") {
+    // decimal32-written-as-64-bit.snappy.parquet was generated using a 3rd-party library. It has
+    // 10 rows of Decimal(9, 1) written as LongDecimal instead of an IntDecimal
+    readParquetFile(testFile("test-data/decimal32-written-as-64-bit.snappy.parquet"), false) {
+      df => assert(10 == df.collect().length)
+    }
+    // decimal32-written-as-64-bit-dict.snappy.parquet was generated using a 3rd-party library. It
+    // has 2048 rows of Decimal(3, 1) written as LongDecimal instead of an IntDecimal
+    readParquetFile(testFile("test-data/decimal32-written-as-64-bit-dict.snappy.parquet"), false) {
+      df => assert(2048 == df.collect().length)
+    }
+  }
+
+  test("SPARK-34167: read LongDecimals with precision < 10, VectorizedReader on") {
+    // decimal32-written-as-64-bit.snappy.parquet was generated using a 3rd-party library. It has
+    // 10 rows of Decimal(9, 1) written as LongDecimal instead of an IntDecimal
+    readParquetFile(testFile("test-data/decimal32-written-as-64-bit.snappy.parquet")) { df =>
```

Comment:
How did you generate this parquet file? The footer contains a metadata entry: `"org.apache.spark.sql.parquet.row.metadata": "{"type":"struct","fields":[{"name":"_c0","type":"decimal(9,1)","nullable":true,"metadata":{}}]}"`, which looks pretty weird as Spark can't generate such parquet files.

Reply:
I used a 3rd-party ([libcudf](https://github.com/rapidsai/cudf)) to generate the parquet file.

Reply:
OK seems it uses Spark under the hood...

Reply:
no, it uses the GPU to write the file.

Reply:
Why does the generated parquet file have the special spark metadata key? Someone else also mentioned it before: https://github.com/rapidsai/cudf/pull/3555#discussion_r355241948

Reply:
We found this while adding in parquet write support for decimal values to https://github.com/NVIDIA/spark-rapids/
It has the Spark metadata because it was written by an incomplete version of the plugin running in Apache Spark.  We have fixed the issue in the plugin so it will no longer write out files like this. It matches what Spark does.  But because it was not technically a violation of the parquet specification we decided to fix Spark so it could read in the files correctly because we know that the python API to RAPIDS is also going to produce parquet files (without the Spark metadata) that also exhibit these conditions because they want to let the user have control over how it is written.